pkg/volume/util/volumepathhandler/volume_path_handler_linux.go

```diff
@@ -93,14 +125,35 @@ func (v VolumePathHandler) RemoveLoopDevice(device string) error {
 	return nil
 }
 
-func parseLosetupOutputForDevice(output []byte) (string, error) {
+func parseLosetupOutputForDevice(output []byte, path string, exactMatch bool) (string, error) {
```

Comment:
I wouldn't mind 2 functions. Maybe parseLosetupAssociatedOutput... (-j) and parseLosetupFileOutput... (-f)

Reply:
Thank you for suggestion.

I was thinking about providing a way to call `DetachFileDevice` that can release descriptor lock regardless of how its backend path looks like (For example, to be used before calling `TearDownDevice` in `operationGenerator` to ensure no descriptor lock remains). However, we would rather avoid such a situation that needs force releasing locks (To do so, we might need to add a check if all descriptor lock is released before calling `UnmapDevice` so that there will be no descriptor lock).  So, let's change `DetachFileDevice` to release the path that exact matches, then we will no longer need the `exactMatch` flag, which is much simpler.

Also, I found that we might need to add to `MapPodDevice` and `UnmapPodDevice` implementations to all in-tree plugins that support block volume so that their device paths to be per pod (I'm thinking about making all in-tree plugins also bind mount their actual devices to per pod paths in `MapPodDevice`). Without them, descriptor lock won't work well for in-tree plugins for multi attaching the same volume case.

Reply:
Do we need the file descriptor lock at all if we are already bind mounting the path /dev/asdf to globalMapPath? It is getting complicated again with the in-tree vs csi case :(

Reply:
The last commit will make in-tree driver also make use of per pod descriptor lock.
As you indicate, this is to just make descriptor lock to be per pod (ref count is already per pod), so if we remove descriptor lock logic, we can remove most of these codes.  

(I haven't tested enough, but just bind mounting device doesn't seem to prevent device from being detached, as loop back device do.)

@msau42 

Do you think that we should remove descriptor lock for simplicity? Or any good ideas to improve?